aws/resource_aws_ssm_parameter_test.go

```diff
@@ -50,6 +50,25 @@ func TestAccAWSSSMParameter_update(t *testing.T) {
 	})
 }
 
+func TestAccAWSSSMParameter_changeNameForcesNew(t *testing.T) {
+	name := acctest.RandString(10)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSSSMParameterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSSSMParameterBasicConfig(name, "bar"),
+			},
+			{
+				Config:             testAccAWSSSMParameterBasicConfigChangeName(name, "baz1"),
+				PlanOnly:           true,
+				ExpectNonEmptyPlan: true,
```

Comment:
I think this test isn't testing the new behaviour. 🤔  (a) It is changing the value, not the name. (b) even if it was changing the name, we'd still need to check existence of the old SSM parameter via API as it gets wiped from state, so `testAccCheckAWSSSMParameterDestroy` wouldn't catch it.